pkg/server/api/entry/v1/service_test.go

```diff
@@ -2893,6 +2896,510 @@ func TestGetAuthorizedEntries(t *testing.T) {
+		err  string
+		code codes.Code
+	}
+
+	for _, tt := range []struct {
+		name              string
+		code              codes.Code
+		fetcherErr        string
+		authorizedEntries []*types.Entry
+		steps             []step
+		expectLogs        []spiretest.LogEntry
+		omitCallerID      bool
+	}{
+		{
+			name:              "success no paging",
+			authorizedEntries: []*types.Entry{entry1, entry2},
+			steps: []step{
+				{
+					req: &entryv1.SyncAuthorizedEntriesRequest{},
+					resp: &entryv1.SyncAuthorizedEntriesResponse{
+						Entries: []*types.Entry{entry1, entry2},
+					},
+				},
+			},
+			expectLogs: []spiretest.LogEntry{
+				{
+					Level:   logrus.InfoLevel,
+					Message: "API accessed",
+					Data: logrus.Fields{
+						telemetry.Status: "success",
+						telemetry.Type:   "audit",
+					},
+				},
+			},
+		},
+		{
+			name:              "success with paging",
+			authorizedEntries: []*types.Entry{entry2, entry3, entry1},
+			steps: []step{
+				// Sends initial request and gets back first page of sparse entries
+				{
+					req: &entryv1.SyncAuthorizedEntriesRequest{},
+					resp: &entryv1.SyncAuthorizedEntriesResponse{
+						EntryRevisions: []*entryv1.EntryRevision{
+							{Id: "entry-2", RevisionNumber: 2},
+							{Id: "entry-3", RevisionNumber: 3},
+						},
+						More: true,
+					},
+				},
+				// Gets back second page of sparse entries
+				{
+					resp: &entryv1.SyncAuthorizedEntriesResponse{
+						EntryRevisions: []*entryv1.EntryRevision{
+							{Id: "entry-1", RevisionNumber: 1},
+						},
+						More: false,
+					},
+				},
+				// Requests all entries and gets back first page of full entries
+				{
+					req: &entryv1.SyncAuthorizedEntriesRequest{
+						Ids: []string{"entry-3", "entry-1", "entry-2"},
+					},
+					resp: &entryv1.SyncAuthorizedEntriesResponse{
+						Entries: []*types.Entry{entry1, entry2},
+						More:    true,
+					},
+				},
+				// Gets back second page of full entries
+				{
+					resp: &entryv1.SyncAuthorizedEntriesResponse{
+						Entries: []*types.Entry{entry3},
+						More:    false,
+					},
+				},
+				// Requests one full page of entries and gets back only page
+				{
+					req: &entryv1.SyncAuthorizedEntriesRequest{
+						Ids: []string{"entry-1", "entry-3"},
+					},
+					resp: &entryv1.SyncAuthorizedEntriesResponse{
+						Entries: []*types.Entry{entry1, entry3},
+						More:    false,
+					},
+				},
+				// Requests less than a page of entries and gets back only page
+				{
+					req: &entryv1.SyncAuthorizedEntriesRequest{
+						Ids: []string{"entry-2"},
+					},
+					resp: &entryv1.SyncAuthorizedEntriesResponse{
+						Entries: []*types.Entry{entry2},
+						More:    false,
+					},
+				},
+				// Requests entry that does not exist
+				{
+					req: &entryv1.SyncAuthorizedEntriesRequest{
+						Ids: []string{"entry-4"},
+					},
+					resp: &entryv1.SyncAuthorizedEntriesResponse{
+						Entries: nil,
+						More:    false,
+					},
+				},
+				// Request nothing
```

Comment:
can you add a test case where you have 3 items, but in the middle one is not found?
that will reasult in a single page of 2
and what happens if the 3rd is not found? (I suspect it will be page with 2 + another page with nil.

Reply:
Sure. Added test cases.